lib/explodePseudoStyles.js

```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+function explodePseudoStyles(style) {
+  var hoverStyles = null;
+  var activeStyles = null;
+  var focusStyles = null;
+
+  for (var name in style) {
+    if (style.hasOwnProperty(name) ) {
+      if (name.startsWith('hover')) {
+        hoverStyles = hoverStyles || {};
+        // Strip the name of the pseudo selector prefix and transform first char
+        // to lowercase to ensure correct markup generation and autoprefixing
+        var prop = name.substr(5);
+        hoverStyles[prop.charAt(0).toLowerCase() + prop.slice(1)] = style[name];
+        // Remove the key from main style entry
+        delete style[name];
```

Comment:
rather than mutating the `style` object that's passed in, should we build up a new `baseStyle` object?